fronts-client/src/services/recipeQuery.ts

```diff
@@ -0,0 +1,167 @@
+import url from '../constants/url';
+import { Recipe } from '../types/Recipe';
+
+interface KeyAndCount {
+  key: string;
+  doc_count: number;
+}
+
+export interface ChefSearchParams {
+  query?: string;
+  limit?: number;
+}
+
+export interface RecipeSearchFilters {
+  diets?: string[];
+  contributors?: string[];
+  filterType: "During" | "Post";
+}
+
+export interface RecipeSearchParams {
+  queryText: string;
+  searchType?: "Embedded"|"Match"|"Lucene",
+  fields?: string[];
+  kfactor?: number;
+  limit?: number;
+  filters?: RecipeSearchFilters;
+}
+
+export interface ChefSearchResponse {
+
+}
+
+export interface RecipeSearchResponse {
+  hits: number;
+  maxScore: number;
+  recipes: RecipeSearchHit[];
+}
+
+interface RecipeSearchTitlesResponse {
+  score: number;
+  href: string;
+}
+
+export type RecipeSearchHit = Recipe & {
+  score: number;
+}
+
+export interface DietSearchResponse {
+  "diet-ids": KeyAndCount[];
```

Comment:
tiny one: can we try naming of params camelCase and not hyphen. not sure though if we need same name as coming in json?

Reply:
Heh, heh..... yeah that is a sticking point here.  We _do_ need the same name that's coming from the JSON and _that_ is coming _directly_ from the Elasticsearch index (it's an aggregation response). I forget the reason it's hyphenated there - I think it probably dates all the way back to Hatch - but we are kinda stuck with it for now I'm afraid 🤷